onnxruntime/core/optimizer/unsqueeze_elimination.cc

```diff
@@ -40,6 +40,10 @@ Status UnsqueezeElimination::Apply(Graph& graph, Node& node, RewriteRuleEffect&
   // Generate new dims.
   InlinedVector<int64_t> new_dims(output_rank, 0);
   for (int64_t axis : axes) {
+    if (static_cast<size_t>(axis) >= new_dims.size() || static_cast<size_t>(axis) < 0) {
```

Comment:
```suggestion
    if (static_cast<size_t>(axis) >= new_dims.size()) {
```

Line 36 converts any negative axis to positive, so it seems like there should be no possible way for the second check to match the `axis` value.

Reply:
Yes I fixed that too